src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/Subscales.utils.ts

```diff
@@ -40,3 +41,162 @@ export const getSubscalesToRender = (
 
   return result;
 };
+
+export const getAllSubscalesToRender = (
+  allSubscalesToRender: SubscaleToRender,
+  item: ActivityCompletion,
+  subscale: ActivitySettingsSubscale,
+  activityItems: Record<string, { answer: AnswerDTO; activityItem: Item }>,
+) => {
+  if (!allSubscalesToRender[subscale.name]) {
+    allSubscalesToRender[subscale.name] = {
+      restScores: {},
+    };
+
+    for (const subscaleItem of subscale.items) {
+      if (subscaleItem.type === ElementType.Item) {
+        const formattedItem = formatActivityItemAnswers(
+          activityItems[subscaleItem.name],
+          item.endDatetime,
+        );
+
+        if (!allSubscalesToRender?.[subscale.name]?.items) {
+          allSubscalesToRender[subscale.name] = {
+            ...allSubscalesToRender[subscale.name],
+            items: [formattedItem],
+          };
+        } else {
+          allSubscalesToRender[subscale.name].items?.push(formattedItem);
+        }
+      } else {
+        allSubscalesToRender[subscale.name] = {
+          ...allSubscalesToRender[subscale.name],
+          restScores: {
+            ...allSubscalesToRender[subscale.name].restScores,
+            [subscaleItem.name]: {},
+          } as Record<string, never>,
+        };
+      }
+    }
+
+    return allSubscalesToRender;
+  }
+
+  for (const subscaleItem of subscale.items) {
+    if (subscaleItem.type === ElementType.Item) {
+      const itemIndex =
+        allSubscalesToRender[subscale.name]?.items?.findIndex(
+          (item) => item.activityItem.name === subscaleItem.name,
+        ) ?? -1;
+
+      if (itemIndex < 0) {
+        return allSubscalesToRender;
+      }
+
+      const { activityItem, answers } = compareActivityItem(
+        allSubscalesToRender[subscale.name].items![itemIndex],
+        activityItems[subscaleItem.name],
+        item.endDatetime,
+      );
+
+      allSubscalesToRender[subscale.name].items![itemIndex] = {
+        activityItem,
+        answers,
+      };
+    }
+  }
+
+  return allSubscalesToRender;
+};
+
+export const formatCurrentSubscales = (currentSubscales: ActivityCompletionToRender) =>
+  Object.keys(currentSubscales).reduce(
+    (formattedSubscales: ActivityCompletionToRender, subscaleName) => {
+      const currItems = currentSubscales[subscaleName]?.items || [];
+      const updatedItems = currItems?.reduce(
+        (items: Record<string, FormattedResponse>, formattedResponse: FormattedResponse) => {
```

Comment:
no need to cast the type
```suggestion
        (items: Record<string, FormattedResponse>, formattedResponse) => {
```